graph/repo.go

```diff
@@ -59,3 +73,28 @@ func TryMakeURI(cloneURL string) (string, error) {
 func URIEqual(a, b string) bool {
 	return strings.EqualFold(a, b)
 }
+
+// Removes VCS part if any, git:http://.. => http://..
+func removeVCSPart(url string) string {
+	parts := strings.SplitN(url, ":", 3)
+	if len(parts) < 3 {
+		// does not look as foo:bar:...
+		return url
+	}
+	for _, r := range parts[0] {
+		if !isASCIILetter(r) {
+			return url
+		}
+	}
+	for _, r := range parts[1] {
+		if !isASCIILetter(r) {
+			return url
+		}
+	}
+	return parts[1] + ":" + parts[2]
+}
+
+// Returns true if given rune is ASCII letter
+func isASCIILetter(r rune) bool {
+	return r >= 'a' && r <= 'z' || r >= 'A' && r <= 'Z'
```

Comment:
Really minor style thought, but I [would write](https://github.com/gopherjs/snippet-store/blob/1f77889ea9974c582661da570ae5c35686763b10/id.go#L31) this as:

```Go
return 'a' <= r && r <= 'z' || 'A' <= r && r <= 'Z'
```

It's up to you which you think is better.